opm/output/eclipse/WriteRFT.hpp

```diff
@@ -0,0 +1,87 @@
+/*
+  Copyright (c) 2019 Equinor ASA
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef OPM_WRITE_RFT_HPP
+#define OPM_WRITE_RFT_HPP
+
+namespace Opm {
+
+    class EclipseGrid;
+    class Schedule;
+    class UnitSystem;
+
+} // namespace Opm
+
+namespace Opm { namespace data {
+
+    class WellRates;
+
+}} // namespace Opm::data
+
+namespace Opm { namespace EclIO { namespace OutputStream {
+
+    class RFT;
+
+}}} // namespace Opm::EclIO::OutputStream
+
+namespace Opm { namespace RftIO {
+
+    /// Collect RFT data and output to pre-opened output stream.
+    ///
+    /// RFT data is output for all affected wells at a given timestep,
+    /// and consists of
+    ///
+    ///   1) Time stamp (elapsed and date)
+    ///   2) Metadata about the output (units of measure, well types
+    ///      data type identifier)
+    ///   3) Depth, pressure, Sw, Sg, (I,J,K), and hostgrid for each
+    ///      reservoir connection of the affected well.
+    ///
+    /// If no RFT output is requested at given timestep, then this
+    /// function returns with no output written to the RFT file.
+    ///
+    /// \param[in] reportStep Report step time point for which to output
+    ///    RFT data.
+    ///
+    /// \param[in] elapsed Number of seconds of simulated time until
+    ///    this report step (\p reportStep).
+    ///
+    /// \param[in] usys Run's active unit system conventions.
```

Comment:
I agree that we have a notion of *active* unit system - but that is really just an implementation detail; for all methods which take a unitsystem argument like the current - you can in principle pass different insantances; i.e. the following should work perfectly OK:
```
RftIO::write(..., METRIC, "METRIC_CASE.RFT");
RftIO::write(..., FIELD, "FIELD_CASE.RFT");
```

Reply:
> for all methods which take a unitsystem argument like the current - you can in principle pass different insantances

Right.  I've updated the parameter description accordingly.

> the following should work perfectly OK:
> 
> ```C++
> RftIO::write(/* ... */, METRIC, "METRIC_CASE.RFT");
> RftIO::write(/* ... */, FIELD, "FIELD_CASE.RFT");
> ```

It does.  We use that in the `test_RFT` unit test.